File: JavaScript/1229-Meeting Scheduler.js
/**
 * @param {number[][]} slots1
 * @param {number[][]} slots2
 * @param {number} duration
 * @return {number[]}
 */
var minAvailableDuration = function(slots1, slots2, duration) {
    let ind1 = 0, ind2 = 0;
    let sorting = (a,b) =>{
        return a[0] - b[0];
    }
    slots1.sort(sorting);
    slots2.sort(sorting);
    
    while (ind1 < slots1.length && ind2 < slots2.length){
        let start = Math.max(slots1[ind1][0], slots2[ind2][0]);
        let end = Math.min(slots1[ind1][1], slots2[ind2][1]);
        if (end-start >= duration){
            return [start, start+duration];
        }
        if (slots1[ind1][1] > slots2[ind2][1]){
            ind2++;
        }else{
            ind1++;
        }
    }
    return [];
};

File: C++/1229-Meeting Scheduler.cpp
class Solution {
public:
    vector<int> minAvailableDuration(vector<vector<int>>& slots1, vector<vector<int>>& slots2, int duration) {
        int i = 0, j = 0;
        sort(slots1.begin(), slots1.end());
        sort(slots2.begin(), slots2.end());
        while (i < slots1.size() && j < slots2.size()) {
            int left = max(slots1[i][0], slots2[j][0]);
            int right = min(slots1[i][1], slots2[j][1]);
            if (right-left >= duration) {
                return {left, left+duration};
            }
            if (slots1[i][1] <= slots2[j][1]) i++;
            else j++;
        }

        return {};
    }
};